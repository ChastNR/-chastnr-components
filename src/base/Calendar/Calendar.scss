.cldr__icon {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  flex: 0 0 auto;

  & > span {
    cursor: pointer;
    color: #9299a2;
    transition: color 0.2s ease;

    &:hover {
      color: #5c636b;
    }

    & > span {
      position: relative;
      display: inline-flex;
      align-items: center;
      justify-content: center;
      vertical-align: top;

      & > span {
        max-width: 100%;
        max-height: 100%;
        position: relative;
        display: inline-block;
        transition-duration: 0.2s;
        transition-timing-function: ease;
        transition-property: background-color, color;
        color: inherit;
        width: 24px;
        height: 24px;
        transform: none;
      }
    }
  }
}
