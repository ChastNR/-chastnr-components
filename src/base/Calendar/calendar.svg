<svg xmlns="http://www.w3.org/2000/svg" focusable="false" viewBox="0 0 24 24"><path d="M19 10H5v8a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-8zm-3-5V4a1 1 0 0 1 2 0v1a3 3 0 0 1 3 3v10a3 3 0 0 1-3 3H6a3 3 0 0 1-3-3V8a3 3 0 0 1 3-3V4a1 1 0 1 1 2 0v1h8zm-9 7h2v2H7v-2zm4 0h2v2h-2v-2zm4 0h2v2h-2v-2zm0 3h2v2h-2v-2zm-4 0h2v2h-2v-2zm-4 0h2v2H7v-2z" fill="currentColor" stroke="none"></path></svg>