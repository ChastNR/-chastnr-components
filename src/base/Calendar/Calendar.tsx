import { memo } from 'react';
import { ReactComponent as CalendarSvg } from './calendar.svg';

import './Calendar.scss';

const Calendar = memo(() => (
  <div className="cldr__icon">
    <span>
      <span>
        <span>
          <CalendarSvg />
        </span>
      </span>
    </span>
  </div>
));

export { Calendar, CalendarSvg };
