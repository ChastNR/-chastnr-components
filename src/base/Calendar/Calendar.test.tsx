import { render } from '@testing-library/react';

import { Calendar } from './Calendar';

describe('<Calendar/>', () => {
  it('should equal snapshot', () => {
    const calendar = render(<Calendar />);

    expect(calendar.container.firstChild).toMatchSnapshot();
  });
});
