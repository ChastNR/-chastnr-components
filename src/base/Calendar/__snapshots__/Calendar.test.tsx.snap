// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar/> should equal snapshot 1`] = `
<div
  class="cldr__icon"
>
  <span>
    <span>
      <span>
        <span>
          content
        </span>
      </span>
    </span>
  </span>
</div>
`;
